systemd_wait.py

```diff
@@ -52,46 +55,53 @@ def wait(bus, unit, target_states):
     )
 
     def get_active_state():
+        """Get the current state of the unit."""
         return unit_int.Get(UNIT_INTERFACE, "ActiveState")
 
     def handler(interface, changed, invalidated):
-        nonlocal state
+        """Handle PropertiesChanged events."""
         if interface == UNIT_INTERFACE:
             if "ActiveState" in invalidated:
                 state = get_active_state()
             elif "ActiveState" in changed:
                 state = changed['ActiveState']
             if state in target_states:
-                event_loop.quit()
+                EVENT_LOOP.quit()
 
-    unit_int.connect_to_signal('PropertiesChanged', handler) 
+    unit_int.connect_to_signal('PropertiesChanged', handler)
 
     manager_int = dbus.Interface(
-        bus.get_object('org.freedesktop.systemd1', '/org/freedesktop/systemd1'),
+        bus.get_object('org.freedesktop.systemd1',
+                       '/org/freedesktop/systemd1'),
         dbus_interface="org.freedesktop.systemd1.Manager"
     )
 
     manager_int.Subscribe()
 
-    state = get_active_state()
-    if state not in target_states:
-        event_loop.run()
+    if get_active_state() not in target_states:
+        EVENT_LOOP.run()
 
     manager_int.Unsubscribe()
 
-    return state
```

Comment:
So, `nonlocal state` is why this used to work. Unfortunately, doing this *this* way:

1. Introduces an additional dbus method call.
2. May *not* return the state that caused us to stop waiting. That means we could return a state to the user that *isn't* in the set of states they told us to wait for.

I'd just leave this as it was (maybe commenting on what we're doing here). I'd take a less "spooky magic at a distance" way of doing this if you can think of one but I can't think of any simple solutions.

Reply:
Oh wow, I wasn't even aware that `nonlocal` was a thing - I initially tried this with python 2.7 and mistakenly assumed it was a typo!  I'll put it back.

Reply:
Done, should be ready for another look.